solarkraft/src/fetcher/storage.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license
+ * [Apache-2.0](https://github.com/freespek/solarkraft/blob/main/LICENSE)
+ */
+
+/**
+ * Storage data for contract invocations.
+ *
+ * Igor Konnov, 2024.
+ */
+
+import { OrderedMap } from 'immutable'
+
+/**
+ * Ordered mapping from field names to their native values (JS),
+ * as produced by Stellar SDK.
+ */
+export type FieldsMap = OrderedMap<string, any>
+
+/**
+ * A storage entry for a performed contract call.
+ */
+export interface ContractCallEntry {
+    /**
+     * The ledger number, which is also called the height or the block number in other systems.
+     */
+    height: number
+    /**
+     * Transaction hash.
+     */
+    txHash: string
+    /**
+     * The address of the contract being called.
+     */
+    contractId: string
+    /**
+     * The name of the method.
+     */
+    method: string
+    /**
+     * The arguments passed to the method, in native JS.
+     */
+    methodArgs: any[]
+    /**
+     * The return value in native JS.
+     */
+    returnValue: any
+    /**
+     * Ordered mapping from field names to their native values (JS).
+     * This mapping contains values only for the fields that have been set
```

Comment:
So if I want all values after the tx, I have to merge `oldFields` and `fields`?

`updatedFields` may be a better name here.

Reply:
No, actually, `fields` contains all fields stored so far. It's basically the new state.

Reply:
Of course, if a contract removes a field from the storage, then `fields` will have fewer fields.

Reply:
I have updated the comments a bit

Reply:
Then I dont understand the comment (`contains values only for the fields that have been set by transactions`). Which fields aren't  set then?

Reply:
Here is the fresh version. It's all modulo my understanding of how Soroban stores the storage in Stellar:

```js
    /**
     * Ordered mapping from field names to their native values (JS).
     * This mapping contains values only for the fields that have been created
     * or updated by a transaction in the past. It may happen that `fields` contains
     * fewer fields than `oldFields`, when the contract deletes some fields
     * from the storage. Also, `fields` may become empty when the storage
     * goes over TTL.
     */
```

Reply:
Oh, you meant values that ever have been set! Understood, lg